pkgs/build-support/go/module.nix

```
@@ -19,10 +19,7 @@
 #
 # if vendorHash is null, then we won't fetch any dependencies and
 # rely on the vendor folder within the source.
-, vendorHash ? "_unset"
-# same as vendorHash, but outputHashAlgo is hardcoded to sha256
-# so regular base32 sha256 hashes work
-, vendorSha256 ? "_unset"
+, vendorHash ? args'.vendorSha256
```

```suggestion
, vendorHash ? args'.vendorSha256 or throw "buildGoModule: Expect vendorHash"
```

The default error message would be "vendorSha256 is missing" without the `or throw`

```
nix-repl> pkgs.buildGoModule { src = pkgs.emptyDirectory; name = "empty"; }
error: attribute 'vendorSha256' missing

       at /data/data/com.termux.nix/files/home/Projects/NixOS/nixpkgs/pkgs/build-support/go/module.nix:22:16:

           21| # rely on the vendor folder within the source.
           22| , vendorHash ? args'.vendorSha256
             |                ^
           23| # Whether to delete the vendor folder supplied with the source.
«derivation
```

Other than that, LGTM!